File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
    int left  = 0  ;
    int  right =  heights.length-1;
    int maxarea =0;

    while(left<=right){
        if(heights[left]>heights[right]){
            maxarea= Math.max(maxarea,heights[right]*(right-left));
            right--;
        }
        else{
            maxarea= Math.max(maxarea,heights[left]*(right-left));
            left++;
        }
    }
    return maxarea;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        Char_set =set()
        maxlen = 0
        left = 0
        for i in range (len(s)):

            while s[i] in Char_set:
                Char_set.remove(s[left])
                left+=1

            Char_set.add(s[i])
            maxlen=max(maxlen,i-left+1)

        return maxlen
                

        